### PR TITLE
v2 wip

### PR DESCRIPTION
**Pull request by tomasstrejcek**

### CODE REVIEW

File: sample/server.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import express from 'express'
+import compression from 'compression'
+// canvas generator
+import { Generator, ImageOptions } from '../src/generator'
+import fs from "fs"
+
+const server = express()
+
+const tmpDir = __dirname + '/tmp/'
+const publicDir = __dirname + '/public/'
```
Comment (cernytomas) [inline sample/server.ts]: unused. pls remove

File: sample/server.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return true
+      }
+
+      // fallback to standard filter function
+      return compression.filter(req, res)
+    }
+  })
+)
+
+server.use(express.static(publicDir))
```
Comment (cernytomas) [inline sample/server.ts]: not needed

File: sample/server.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+server.use(express.static(tmpDir))
+
+server.get('/file', (req, res) => {
+  const options: ImageOptions = {
+    height: 100, width: 300, bg: '00FF00', color: 'FF0000', frames: 60, fontSize: 48
+  }
+
+  const tmpDir = process.cwd() + '/tmp/'
+  if (!fs.existsSync(tmpDir)) {
+    fs.mkdirSync(tmpDir)
```
Comment (cernytomas) [inline sample/server.ts]: create directory at the server start and die server if permission required

File: sample/server.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  }
+  const filePath = tmpDir + req.query.name + '.gif'
+  const fileStream = fs.createWriteStream(filePath)
+
+  new Generator(options)
```
Comment (cernytomas) [inline sample/server.ts]: response is missing

File: sample/server.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+server.get('/file', (req, res) => {
+  const options: ImageOptions = {
+    height: 100, width: 300, bg: '00FF00', color: 'FF0000', frames: 60, fontSize: 48
+  }
+
+  const tmpDir = process.cwd() + '/tmp/'
+  if (!fs.existsSync(tmpDir)) {
+    fs.mkdirSync(tmpDir)
+  }
+  const filePath = tmpDir + req.query.name + '.gif'
```
Comment (cernytomas) [inline sample/server.ts]: assert `name` parameter for empty input

File: sample/server.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    height: 100, width: 300, bg: '00FF00', color: 'FF0000', frames: 60, fontSize: 36, fontFamily: 'Arial'
+  }
+  new Generator(options)
+    .setTimer(req.query.end || new Date(), req.query.timezone || 'UTC')
+    .setOutputStream(res)
+    .encode()
+})
+
+server.listen(process.env.PORT || 3000, function () {
+  console.log('Express server listening on http://localhost:%d/server in %s mode', this.address().port, server.settings.env)
```
Comment (cernytomas) [inline sample/server.ts]: `/server` is not a valid path

File: sample/server.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  const fileStream = fs.createWriteStream(filePath)
+
+  new Generator(options)
+    .setTimer(req.query.end || new Date(), req.query.timezone || 'UTC')
+    .setOutputStream(fileStream)
+    .encode()
+})
+
+server.get('/stream', (req, res) => {
+  const options: ImageOptions = {
```
Comment (cernytomas) [inline sample/server.ts]: should be nice to have possibility to set these options as request parameters

File: README.md [hunk 8]
```diff
@@ -1,33 +1,30 @@
...
+# Countdown GIF generator for emails and whatnot
 
-The very simple app I have created allows you to generate a countdown timer animated gif depending on the URL parameters you provide. [View demo](https://date-gif.herokuapp.com/).
+Simple library to generate countdown GIFs in nodejs using Cairo/Canvas implementation usable for on-demand image generation.
 
-## URL Parameters (*required)
-
-* **time*** - Date &amp; time when your countdown will end [e.g. 2016-06-24T20:35]
+## Image Options
 * **frames** - number of frames (also number of seconds) the countdown will run before looping [defaults to 30]
```
Comment (cernytomas) [inline README.md]: it's not possible to set these parameters

File: sample/server.ts (hunk 5 above)
Comment (tomasstrejcek) [reply sample/server.ts]: its gonna be undefined which is fine

File: sample/server.ts (hunk 7 above)
Comment (tomasstrejcek) [reply sample/server.ts]: server is just sample, marginally you absolutely dont want to set these from outside, but via some predefined config options, its very easy to cause ddos via image generation

File: sample/server.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
+// server
+import express from 'express'
+import compression from 'compression'
+// canvas generator
+import { Generator, ImageOptions } from '../src/generator'
+import fs from "fs"
```
Comment (cernytomas) [inline sample/server.ts]: double quotes

**Pull request merged**